app.js

```diff
@@ -192,64 +192,39 @@ app.get('/oops', function(req, res, next) {
 
 app.get('/status', (req, res, next) => res.send('CLSI is alive\n'))
 
-const resCacher = {
-  contentType(setContentType) {
-    this.setContentType = setContentType
-  },
-  send(code, body) {
-    this.code = code
-    this.body = body
-  },
-
-  // default the server to be down
-  code: 500,
-  body: {},
-  setContentType: 'application/json'
-}
-
-let shutdownTime
+let PROCESS_IS_TOO_OLD = false
 if (Settings.processLifespanLimitMs) {
   Settings.processLifespanLimitMs +=
     Settings.processLifespanLimitMs * (Math.random() / 10)
-  shutdownTime = Date.now() + Settings.processLifespanLimitMs
-  logger.info('Lifespan limited to ', shutdownTime)
-}
+  logger.info(
+    'Lifespan limited to ',
+    Date.now() + Settings.processLifespanLimitMs
+  )
 
-const checkIfProcessIsTooOld = function(cont) {
-  if (shutdownTime && shutdownTime < Date.now()) {
+  setTimeout(() => {
     logger.log('shutting down, process is too old')
-    resCacher.send = function() {}
-    resCacher.code = 500
-    resCacher.body = { processToOld: true }
-  } else {
-    cont()
-  }
+    PROCESS_IS_TOO_OLD = true
+  }, Settings.processLifespanLimitMs)
 }
 
 if (Settings.smokeTest) {
-  const runSmokeTest = function() {
-    checkIfProcessIsTooOld(function() {
-      logger.log('running smoke tests')
-      smokeTest.run(
-        require.resolve(__dirname + '/test/smoke/js/SmokeTests.js')
-      )({}, resCacher)
-      return setTimeout(runSmokeTest, 30 * 1000)
+  function runSmokeTest() {
+    if (PROCESS_IS_TOO_OLD) return
+    logger.log('running smoke tests')
+    smokeTest.triggerRun(err => {
+      logger.error({ err }, 'smoke tests failed')
```

Comment:
This gets logged every time, maybe make it conditional on `if(err)`.

Reply:
Right, I remember to having fixed that -- https://github.com/das7pad/clsi-sharelatex/commit/23e52cf7b76a7b1195e89760edde24c74259f872